### PR TITLE
core: doc: add note for countNonZero, hasNonZero and findNonZero

### LINKED ISSUE #25345: cv::countNonZero /documentation/ add remark about using cv::hasNonZero
### Describe the doc issue

countNonZero in most cases is used to know if hasNonZero.
Using hasNonZero in that cases is
- more efficient
- semantically correct

### Fix suggestion

To promote adaequate code I suggest to add to the doc of countNonZero
something like:

For determing IF there is a non-zero array element use hasNonZero.
(For multi-channel arrays do a Mat::reshape (O(1) operation) before.

---------

Moreover add links to hasNonZero and Mat::reshape.

### PR DESCRIPTION
**Pull request by Kumataro**
Close #25345 

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by opencv-alalek** [comment]
Documentation preview: https://pullrequest.opencv.org/buildbot/export/pr/25356/docs/

@nji9nji9 FYI

File: modules/core/include/opencv2/core.hpp [hunk 1]
```diff
@@ -585,17 +585,37 @@ CV_EXPORTS_AS(sumElems) Scalar sum(InputArray src);
...
 
 /** @brief Counts non-zero array elements.
 
 The function returns the number of non-zero elements in src :
 \f[\sum _{I: \; \texttt{src} (I) \ne0 } 1\f]
+
+The function do not work with multi-channel arrays. If you need to count non-zero array
+elements across all the channels, use Mat::reshape first to reinterpret the array as
+single-channel. Or you may extract the particular channel using either extractImageCOI , or
+mixChannels , or split .
```
Comment (asmorkalov) [inline modules/core/include/opencv2/core.hpp:611]: No need in space before , and . Here and bellow.

Comment (Kumataro) [reply modules/core/include/opencv2/core.hpp]: Thank you for your comment, these lines comes from other functions (e.g. normalize, minMaxIdx).  If *NonZero() functions are need to fix, I think they should be fixed too.

https://github.com/opencv/opencv/blob/e80500828c2f84ce6b46fe1e41744a32ed1738e2/modules/core/include/opencv2/core.hpp#L831-L834

**Review by asmorkalov — APPROVED** [review]
:+1:

**Pull request merged**